### PR TITLE
Add integration test for Services Finalizes Racing with the Repair Loop

### PR DESCRIPTION
**Pull request by aojea**
/kind test
/sig network

Add an integration test that checks that the Services repair loop doesn't conflict with the delete process of a Service with a finalizer

```release-note
NONE
```

### CODE REVIEW

**Comment by aojea** [comment]
/assign @thockin @liggitt 

I'm "resurrecting" this PR after Tim's refactor of the Service logic https://github.com/kubernetes/kubernetes/pull/96684

The original issue was solved and the new Service REST code has much more coverage, but this is the only test that covers the interaction with the repair loop.

File: test/integration/network/services_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	t.Logf("Service created successfully: %v", svc)
+
+	// Delete service
+	if err := client.CoreV1().Services(metav1.NamespaceDefault).Delete(context.TODO(), svcNodePort.Name, metav1.DeleteOptions{}); err != nil {
+		t.Errorf("unexpected error deleting service: %v", err)
+	}
+	t.Logf("Deleted service: %s", svcNodePort.Name)
+
+	// wait for the repair loop to recover the deleted resources
+	time.Sleep(cfg.ExtraConfig.RepairServicesInterval)
```
Comment (thockin) [inline test/integration/network/services_test.go:104]: This feels like it could flake - should you 2x it ?

**Review by thockin — COMMENTED** [review]
LGTM overall - maybe you can clarify in comments which race(s) with the repair loop are being proven here?

File: test/integration/network/services_test.go (hunk 1 above)
Comment (aojea) [reply test/integration/network/services_test.go:104]: I've added 1 second and it works fine, we should not make these takes take too much time, it takes 15 sec now 
```
real    0m15,014s
```

```
stress ./network.test 
5s: 0 runs so far, 0 failures
10s: 0 runs so far, 0 failures
15s: 12 runs so far, 0 failures
20s: 24 runs so far, 0 failures
25s: 24 runs so far, 0 failures
30s: 35 runs so far, 0 failures
35s: 48 runs so far, 0 failures
40s: 50 runs so far, 0 failures
45s: 58 runs so far, 0 failures
50s: 72 runs so far, 0 failures
55s: 74 runs so far, 0 failures
1m0s: 86 runs so far, 0 failures
```

Comment (thockin) [reply test/integration/network/services_test.go]: You could make the interval 2 or 3 seconds?

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**